.github/workflows/GPU.yml

```diff
@@ -0,0 +1,42 @@
+name: GPU
+
+on:
+  pull_request:
+    branches:
+      - master
+      - stable
+
+jobs:
+  build:
+    runs-on: [self-hosted, GPU, benchmark]
+
+    steps:
+      - name: cleanup
+        run: rm -rf *
+      - name: Checkout votca repo
+        uses: actions/checkout@v2
+        with:
+          repository: votca/votca
+          submodules: true
+          ref: ${{ github.base_ref }}
```

Comment:
Problem with that is: "This property is only available when the event that triggers a workflow run is either pull_request or pull_request_target.", so it will fail when triggered by a push to master.

I think you will need to a little magic like in the prepare step:
```bash
if [[ ${GITHUB_REF} = refs/pull/*/merge ]]; then # pull request
  ref="${GITHUB_BASE_REF}"
elif [[ ${GITHUB_REF} = refs/heads/stable ]]; then # stable branch
  ref="stable"
else
  ref="master"
fi
echo "::set-output name=ref::${ref}"
```

And the use `${{ steps.prepare.outputs.ref }}` here.